### PR TITLE
map text and main menu stuff

### PR DESCRIPTION
**Pull request by 2Tie**
mostly disassembling of bank 0 and bank 1. also has what i did to get the build to work, because other team members told me to include it.

### CODE REVIEW

File: home/talk_to_npc.asm [hunk 1]
```diff
@@ -144,14 +145,233 @@ Function30e8:: ; 00:30e8
...
+	ld b, 04 
+	ld c, $12 
+	call DrawTextBox
+	call WaitBGMap
+	call LoadFonts_NoOAMUpdate
+	pop af
+	call Bankswitch
+	ret
+
+TextboxCleanup ; 00:3171
```
Comment (mid-kid) [inline home/talk_to_npc.asm]: Missing colon.

File: home/talk_to_npc.asm [hunk 2]
```diff
@@ -144,14 +145,233 @@ Function30e8:: ; 00:30e8
...
+	ldh [hBGMapMode], a
+	ld a, $90
+	ldh [hWY], a
+	call Function318f 
+	ld hl, wToolgearFlags
+	res 7, [hl]
+	call InitToolgearBuffer
+	ret
+
+Function318f ; 00:318f
```
Comment (mid-kid) [inline home/talk_to_npc.asm]: Missing colon.

File: home/talk_to_npc.asm [hunk 3]
```diff
@@ -144,14 +145,233 @@ Function30e8:: ; 00:30e8
...
+.SetRight ; 322C
+	ld b, a
+	ld c, RIGHT
+	scf
+	ret
+.Reset ; 3231
+	and a
+	ret
+	
+CheckBPressedDebug ; 3233
```
Comment (mid-kid) [inline home/talk_to_npc.asm]: Missing colon.

File: home/talk_to_npc.asm (hunk 1 above)
Comment (2Tie) [reply home/talk_to_npc.asm]: fixed in latest commit

File: home/talk_to_npc.asm (hunk 2 above)
Comment (2Tie) [reply home/talk_to_npc.asm]: fixed in latest commit

File: home/talk_to_npc.asm (hunk 3 above)
Comment (2Tie) [reply home/talk_to_npc.asm]: fixed in latest commit

**Pull request merged**